controllers/front/validation.php

```diff
@@ -66,66 +68,45 @@ public function postProcess()
-            $this->errors[] =
-                $this->module->l('This payment method is not available.', self::FILENAME);
+            $this->errors[] = $this->module->l('This payment method is not available.', self::FILENAME);
             $this->redirectWithNotifications($redirectLink);
         }
 
-        $customer = new Customer($cart->id_customer);
-        if (!Validate::isLoadedObject($customer)) {
-            Tools::redirect($redirectLink);
+        if (Order::getOrderByCartId($this->context->cart->id)) {
+            $this->errors[] = $this->module->l('Order already exists.', self::FILENAME);
+            $this->redirectWithNotifications($redirectLink);
         }
 
-        $currency = $this->context->currency;
-        $total = (float) $cart->getOrderTotal();
+        try {
+            /** @var PaymentFrontController $paymentFrontController */
+            $paymentFrontController = $this->module->getService(PaymentFrontController::class);
 
-        $orderId = Order::getOrderByCartId($cart->id);
-        if (!$orderId) {
-            $this->module->validateOrder(
-                $cart->id,
-                Configuration::get(SaferPayConfig::SAFERPAY_ORDER_STATE_CHOICE_AWAITING_PAYMENT),
-                $total,
+            $initializeResponse = $paymentFrontController->create(
+                $this->context->cart,
                 $paymentMethod,
-                null,
-                [],
-                (int) $currency->id,
-                false,
-                $customer->secure_key
+                (int) Tools::getValue(SaferPayConfig::IS_BUSINESS_LICENCE)
             );
-        }
 
-        /** @var SaferPayInitialize $initializeService */
-        $initializeService = $this->module->getService(SaferPayInitialize::class);
-        try {
-            $isBusinessLicence = Tools::getValue(SaferPayConfig::IS_BUSINESS_LICENCE);
-            $initializeBody = $initializeService->initialize($paymentMethod, $isBusinessLicence);
-        } catch (SaferPayApiException $e) {
+            $redirectLink = $paymentFrontController->getRedirectionUrl($initializeResponse);
+
+            Tools::redirect($redirectLink);
+        } catch (\Exception $exception) {
             /** @var SaferPayExceptionService $exceptionService */
             $exceptionService = $this->module->getService(SaferPayExceptionService::class);
-            $this->errors[] = $exceptionService->getErrorMessageForException($e, $exceptionService->getErrorMessages());
+            $this->errors[] = $exceptionService->getErrorMessageForException($exception, $exceptionService->getErrorMessages());
+
             $redirectLink = $this->context->link->getModuleLink(
                 $this->module->name,
                 'fail',
                 [
                     'cartId' => $this->context->cart->id,
-                    'orderId' => Order::getOrderByCartId($this->context->cart->id),
+                    'orderId' => Order::getIdByCartId($this->context->cart->id),
```

Comment:
Pretty sure one of these are deprecated and other is supported only in later versions, try to check if it exists inside our lowest supported versions of PS

Reply:
these two functions are used over the project not sure which one is compatible with our logic